src/modules/fundingManager/oracle/LM_ManualExternalPriceSetter_v1.sol

```diff
@@ -0,0 +1,181 @@
+    "src/modules/fundingManager/oracle/interfaces/ILM_ManualExternalPriceSetter_v1.sol";
+import {Module_v1} from "src/modules/base/Module_v1.sol";
+import {IOrchestrator_v1} from
+    "src/orchestrator/interfaces/IOrchestrator_v1.sol";
+import {IERC20Metadata} from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+
+/**
+ * @title   Manual External Price Oracle Implementation
+ *
+ * @notice  This contract provides a manual price feed mechanism for token
+ *          operations, allowing authorized users to set and update prices for
+ *          both issuance (buying) and redemption (selling) operations.
+ *
+ * @dev     This contract inherits functionalities from:
+ *              - Module_v1
+ *          The contract maintains two separate price feeds:
+ *              1. Issuance price for token minting/buying
+ *              2. Redemption price for token burning/selling
+ *          Both prices are manually set by the contract owner and must be
+ *          non-zero values.
+ *
+ * @custom:security-contact security@inverter.network
+ *                          In case of any concerns or findings, please refer to
+ *                          our Security Policy at security.inverter.network or
+ *                          email us directly!
+ *
+ * @author  Zealynx Security
+ */
+contract LM_ManualExternalPriceSetter_v1 is
+    ILM_ManualExternalPriceSetter_v1,
+    IOraclePrice_v1,
+    Module_v1
+{
+    //--------------------------------------------------------------------------
+    // Constants
+
+    bytes32 public constant PRICE_SETTER_ROLE = "PRICE_SETTER_ROLE";
+    uint8 private constant INTERNAL_DECIMALS = 18;
+
+    //--------------------------------------------------------------------------
+    // State Variables
+
+    /// @notice The price for issuing tokens (normalized to INTERNAL_DECIMALS)
+    uint256 private _issuancePrice;
+
+    /// @notice The price for redeeming tokens (normalized to INTERNAL_DECIMALS)
+    uint256 private _redemptionPrice;
+
+    /// @notice Decimals of the input token
+    uint8 private _inputTokenDecimals;
+
+    /// @notice Decimals of the output token
+    uint8 private _outputTokenDecimals;
+
+    /// @dev Storage gap for upgradeable contracts
+    uint[50] private __gap;
+
+    //--------------------------------------------------------------------------
+    // Initialization
+
+    /// @inheritdoc Module_v1
+    function init(
+        IOrchestrator_v1 orchestrator_,
+        Metadata memory metadata_,
+        bytes memory configData_
+    ) external override(Module_v1) initializer {
+        __Module_init(orchestrator_, metadata_);
+
+        // Decode input and output token addresses from configData_
+        (address inputToken, address outputToken) = abi.decode(
+            configData_,
+            (address, address)
+        );
+
+        // Store token decimals for price normalization
+        _inputTokenDecimals = IERC20Metadata(inputToken).decimals();
```

Comment:
Could you lead me through what is meant by input/output token and how this should be contextualized.

I see that it is a solution for the conversion but I'm not sure how it should be used. I think this also feeds into the questions of in which token is the return price for issuance/redemption denominated

Reply:
I see, maybe the naming was not ideal.

I've changed them to: 
```
    /// @notice Decimals of the collateral token (e.g., USDC with 6 decimals)
    /// @dev This is the token used to pay/buy with
    uint8 private _collateralTokenDecimals; // previously inputTokenDecimals

    /// @notice Decimals of the issuance token (e.g., ISS with 18 decimals)
    /// @dev This is the token being bought/sold
    uint8 private _issuanceTokenDecimals; // previously outputTokenDecimals
```